docs/docs/tutorial/tpch.md

```diff
@@ -159,12 +159,579 @@ LOAD DATA LOCAL INFILE '/home/username/workspace/tpch1g/lineitem/lineitem.tbl' I
+  "o_comment"        VARCHAR(79),
+  "o_dummy"          VARCHAR(10),
+  PRIMARY KEY ("o_orderkey"));
+
+-- lineitem
+CREATE TABLE IF NOT EXISTS "tpch1g"."lineitem"(
+  "l_orderkey"    INT,
+  "l_partkey"     INT,
+  "l_suppkey"     INT,
+  "l_linenumber"  INT,
+  "l_quantity"    DECIMAL(15,2),
+  "l_extendedprice"  DECIMAL(15,2),
+  "l_discount"    DECIMAL(15,2),
+  "l_tax"         DECIMAL(15,2),
+  "l_returnflag"  CHAR(1),
+  "l_linestatus"  CHAR(1),
+  "l_shipdate"    DATE,
+  "l_commitdate"  DATE,
+  "l_receiptdate" DATE,
+  "l_shipinstruct" CHAR(25),
+  "l_shipmode"    CHAR(10),
+  "l_comment"     VARCHAR(44),
+  "l_dummy"       VARCHAR(10));
+```
+
+### Load Data
+For Redshift, we use Java method to help inserting our data into Redshift tables. `schema` is the Redshift schema you create your TPC-H tables
+and `table` is the table name of TPCH-table, such as `nation`, `region`. `conn` is the Connection class you get from connecting your Redshift cluster using Redshift JDBC driver.
+Suppose your work directory is `/home/username/workspace` and the tpch1g data is stored in `/home/username/workspace/tpch1g`.
+```java
+static void loadRedshiftData(String schema, String table, Connection conn)
+      throws IOException {
+    Statement stmt = conn.createStatement();
+    String concat = "";
+    File file =
+        new File(String.format("/home/username/workspace/tpch1g/%s/%s.tbl", table, table));
+    ResultSet columnMeta =
+        stmt.execute(
+            String.format(
+                "select data_type, ordinal_position from INFORMATION_SCHEMA.COLUMNS where table_name='%s' and table_schema='%s'",
+                table, schema));
+    List<Boolean> quotedNeeded = new ArrayList<>();
+    for (int i = 0; i < columnMeta.getRowCount(); i++) {
+      quotedNeeded.add(true);
+    }
+    while (columnMeta.next()) {
+      String columnType = columnMeta.getString(1);
+      int columnIndex = columnMeta.getInt(2);
+      if (columnType.equals("integer") || columnType.equals("numeric")) {
+        quotedNeeded.set(columnIndex - 1, false);
+      }
+    }
+
+    String content = Files.toString(file, Charsets.UTF_8);
+    for (String row : content.split("\n")) {
+      String[] values = row.split("\\|");
+      row = "";
+      for (int i = 0; i < values.length - 1; i++) {
+        if (quotedNeeded.get(i)) {
+          row = row + getQuoted(values[i]) + ",";
+        } else {
+          row = row + values[i] + ",";
+        }
+      }
+      row = row + "''";
+      if (concat.equals("")) {
+        concat = concat + "(" + row + ")";
+      } else concat = concat + "," + "(" + row + ")";
+    }
+    stmt.execute(String.format("insert into \"%s\".\"%s\" values %s", schema, table, concat));
+  }
+```
 
 ## Cloudera Impala
 
+### Put data to HDFS
+
+Use following commands to put data into HDFS. Suppose the tpch1g data is stored in `/home/username/workspace/tpch1g` and you hope to put your data in `/tmp/tpch1g` in HDFS.
+```bash
+$ sudo su hdfs
```

Comment:
This `sudo` should not be necessary since the files are going to the tmp directory. Please double-check.